### ISSUE TITLE
Path opt is being ignored

### ISSUE THREAD

**Issue by rcoedo** [screepers/node-screeps-api#26]
I've been trying this library to push code to the PTR, but the `path` option is being ignored.

I think I've located the bug in `RawAPI.js`, line 300. The `url` package is probably not working as intended since it is ignoring the relative path in the first parameter.

I think we can work around this issue appending the opts.path to the second parameter of the resolve function.

**Comment by Hiryus** [comment]
Hi rcoedo,

`opts.url` should include the PTR path. It is computed by `setServer` (https://github.com/screepers/node-screeps-api/blob/master/src/RawAPI.js#L269) from the provided options.

Could you give an example of the code you are using ?

Regards,
Hiryus.

**Comment by Hiryus** [comment]
Indeed, `resolveUrl()` is not working as intended: I thought that it was not relative because of the leading slash in front of `/api`, but it still doesn't work without... weird.

**Comment by Hiryus** [comment]
@ags131, we could solve the problem easily with node v7/8 new URL constructor https://nodejs.org/api/url.html#url_constructor_new_url_input_base. Something like this:
``` Javascript
    let url = new URL(path, this.opts.url)
    if (method === 'GET') {
      url.search = querystring.stringify(body)
    }
    if (method === 'POST') {
      opts.headers['content-type'] = 'application/json'
      opts.body = JSON.stringify(body)
    }
    let res = await fetch(url.toString(), opts)
```
... but it requires to upgrade to a non-LTS version.

Another solution would be to standardize `path` to never start with a slash and `opts.path` to always finish with a slash (else, `url.resolve()` doesn't behave properly). Something like 
``` Javascript
      this.opts.url = format(this.opts)
      if (!this.opts.url.endsWith('/')) this.opts.url += '/'
```
in `setServer()` probably.

... or use @rcoedo solution, but it looks less clean (using `url.resolve()` to only resolve part of the path...).

Any thoughts ?

**Comment by Hiryus** [comment]
Referencing pull request. #27

**Comment by ags131** [comment]
Fixed in #27

**Comment by rcoedo** [comment]
Great, thank you all! 🍻
